### ISSUE TITLE
tach report-external [PATH] --raw non deterministic behaviour 

### ISSUE THREAD

**Issue by fraser-langton** [gauge-sh/tach#370]
`tach report-external [PATH] --raw`
```
(venv)> tach report-external common/blueprinter --raw           
fields
processes
jinja2
base
-hypothesis-ftz-detector
json-schema-for-humans
yaml
requests-2-31-0-dist-info
sections
typing-extensions-4-8-0-dist-info
blueprinter
shapely-2-0-6-dist-info
funcs
pytest
pydantic-2-9-2-dist-info
click-8-1-7-dist-info
setuptools-command-build
structlog-22-3-0-dist-info
requests-oauthlib
(venv)> tach report-external common/blueprinter --raw
processes
requests
jinja2
funcs
blueprinter
click
shapely-2-0-6-dist-info
requests-oauthlib
base
hypothesis
yaml
typing-extensions-4-8-0-dist-info
-
sections
fields
pkg-resources
pydantic-2-9-2-dist-info
json-schema-for-humans
structlog
(venv)
> tach report-external common/blueprinter --raw
jinja2
setuptools-dist
pydantic-2-9-2-dist-info
sections
json-schema-for-humans
blueprinter
-
-hypothesis-pytestplugin
shapely-libs
processes
base
fields
requests-2-31-0-dist-info
typing-extensions-4-8-0-dist-info
structlog-22-3-0-dist-info
click
funcs
requests-oauthlib
yaml
```

`tach report-external [PATH]` (removed my paths)
```
Import 'yaml' from package 'yaml'
Import 'shapely.geometry.base.BaseGeometry' from package 'shapely'
Import 'setuptools.setup' from package 'setuptools-75-1-0-dist-info'
Import 'click' from package 'click-8-1-7-dist-info'
Import 'json_schema_for_humans.generate.generate_from_schema' from package 'json-schema-for-humans'
Import 'json_schema_for_humans.generation_configuration.GenerationConfiguration' from package 'json-schema-for-humans'
Import 'requests' from package 'requests'
Import 'structlog' from package 'structlog'
Import 'click' from package 'click-8-1-7-dist-info'
Import 'click' from package 'click-8-1-7-dist-info'
Import 'click' from package 'click-8-1-7-dist-info'
Import 'yaml' from package 'yaml'
Import 'requests' from package 'requests'
Import 'yaml' from package 'yaml'
Import 'requests_oauthlib.OAuth2Session' from package 'requests-oauthlib'
Import 'pydantic.BaseModel' from package 'pydantic-2-9-2-dist-info'
Import 'pydantic.ConfigDict' from package 'pydantic-2-9-2-dist-info'
Import 'pydantic.Field' from package 'pydantic-2-9-2-dist-info'
Import 'pydantic.create_model' from package 'pydantic-2-9-2-dist-info'
Import 'pydantic.model_validator' from package 'pydantic-2-9-2-dist-info'
Import 'pydantic.validator' from package 'pydantic-2-9-2-dist-info'
Import 'yaml' from package 'yaml'
Import 'pydantic.Field' from package 'pydantic-2-9-2-dist-info'
Import 'pydantic.Field' from package 'pydantic-2-9-2-dist-info'
Import 'base' from package 'base'
Import 'funcs' from package 'funcs'
Import 'processes' from package 'processes'
Import 'sections' from package 'sections'
Import 'pydantic.Field' from package 'pydantic-2-9-2-dist-info'
Import 'base' from package 'base'
Import 'pydantic.Field' from package 'pydantic-2-9-2-dist-info'
Import 'pydantic.RootModel' from package 'pydantic-2-9-2-dist-info'
Import 'pydantic.model_validator' from package 'pydantic-2-9-2-dist-info'
Import 'base' from package 'base'
Import 'jinja2' from package 'jinja2'
Import 'pydantic.ConfigDict' from package 'pydantic-2-9-2-dist-info'
Import 'pydantic.Field' from package 'pydantic-2-9-2-dist-info'
Import 'pydantic.RootModel' from package 'pydantic-2-9-2-dist-info'
Import 'pydantic.model_validator' from package 'pydantic-2-9-2-dist-info'
Import 'typing_extensions.Annotated' from package 'typing-extensions-4-8-0-dist-info'
Import 'base' from package 'base'
Import 'pydantic.ConfigDict' from package 'pydantic-2-9-2-dist-info'
Import 'pydantic.Field' from package 'pydantic-2-9-2-dist-info'
Import 'base' from package 'base'
Import 'fields' from package 'fields'
Import 'pydantic.Field' from package 'pydantic-2-9-2-dist-info'
Import 'base' from package 'base'
Import 'fields' from package 'fields'
Import 'funcs' from package 'funcs'
Import 'pydantic.Field' from package 'pydantic-2-9-2-dist-info'
Import 'typing_extensions.Annotated' from package 'typing-extensions-4-8-0-dist-info'
Import 'base' from package 'base'
Import 'fields' from package 'fields'
Import 'pydantic.Field' from package 'pydantic-2-9-2-dist-info'
Import 'base' from package 'base'
Import 'fields' from package 'fields'
Import 'pydantic.BaseModel' from package 'pydantic-2-9-2-dist-info'
Import 'pydantic.ConfigDict' from package 'pydantic-2-9-2-dist-info'
Import 'pydantic.Field' from package 'pydantic-2-9-2-dist-info'
Import 'pydantic.field_validator' from package 'pydantic-2-9-2-dist-info'
Import 'base' from package 'base'
Import 'fields' from package 'fields'
Import 'base' from package 'base'
Import 'fields' from package 'fields'
Import 'base' from package 'base'
Import 'fields' from package 'fields'
Import 'pydantic.BaseModel' from package 'pydantic-2-9-2-dist-info'
Import 'pydantic.Field' from package 'pydantic-2-9-2-dist-info'
Import 'pydantic' from package 'pydantic-2-9-2-dist-info'
Import 'pytest' from package 'pytest-7-0-1-dist-info'
Import 'click' from package 'click-8-1-7-dist-info'
Import 'pytest' from package 'pytest-7-0-1-dist-info'
Import 'click.testing.CliRunner' from package 'click-8-1-7-dist-info'
Import 'pytest' from package 'pytest-7-0-1-dist-info'
Import 'pydantic' from package 'pydantic-2-9-2-dist-info'
Import 'pytest' from package 'pytest-7-0-1-dist-info'
Import 'hypothesis' from package '-hypothesis-ftz-detector'
Import 'pydantic' from package 'pydantic-2-9-2-dist-info'
Import 'pytest' from package 'pytest-7-0-1-dist-info'
Import 'pydantic.ValidationError' from package 'pydantic-2-9-2-dist-info'
Import 'pydantic' from package 'pydantic-2-9-2-dist-info'
Import 'pytest' from package 'pytest-7-0-1-dist-info'
Import 'pydantic.ValidationError' from package 'pydantic-2-9-2-dist-info'
Import 'pydantic' from package 'pydantic-2-9-2-dist-info'
Import 'pytest' from package 'pytest-7-0-1-dist-info'
Import 'pydantic' from package 'pydantic-2-9-2-dist-info'
Import 'pytest' from package 'pytest-7-0-1-dist-info'
Import 'pydantic' from package 'pydantic-2-9-2-dist-info'
Import 'pytest' from package 'pytest-7-0-1-dist-info'
```

**Comment by fraser-langton** [comment]
these are the actual requirements
```
[metadata]
name = blueprinter
version = 0.0.1

[options]
python_requires = ==3.8.*
packages = find:
install_requires =
    pydantic==2.*
    pyyaml==6.*
    requests
    requests-oauthlib>=2
    shapely==2.*

[options.extras_require]
tests =
    hypothesis==6.*
    pytest-asyncio==0.23.*
    pytest-cov
    pytest-mock
    pytest==7.0.*
    testcontainers==3.*
cli =
    click==8.*
    json-schema-for-humans==1.*

[options.packages.find]
exclude = *
```

**Comment by emdoyle** [comment]
These issues should be fixed in the latest release: `0.14.0`

Let me know if it's working for you! @fraser-langton

**Comment by fraser-langton** [comment]
Looks better! But it should be picking up internal modules right?

```
> tach report-external common/blueprinter --raw
click
hypothesis
jinja2
json-schema-for-humans
pydantic
pytest
pyyaml
requests
requests-oauthlib
setuptools
shapely
structlog
```
